src/de.rs

```diff
@@ -563,6 +563,8 @@ impl<'de> de::Deserializer<'de> for Deserializer {
     fn deserialize_byte_buf<V: de::Visitor<'de>>(self, visitor: V) -> Result<V::Value> {
         if let Some(bytes) = self.as_bytes() {
             visitor.visit_byte_buf(bytes)
+        } else if let Some(arr) = self.value.dyn_ref::<Array>() {
+            visitor.visit_seq(SeqDeserializer::new(arr.iter().map(Deserializer::from)))
```

Comment:
Can you extract this into a private helper on Deserializer so that it's reused both here and in `deserialize_seq`? It's kind of small, but tricky enough to deserve a separate function.